url1a745d67fc_sobercheg_InterviewTraining_tgz-pJ8-leetcode_oj_Solution2TestJ8/src/careercup/google/SubarrayDiff.java

```diff
@@ -0,0 +1,120 @@
+package careercup.google;
+
+/**
+ * Created by sobercheg on 11/22/13.
+ * <p/>
+ * http://www.careercup.com/question?id=19286747
+ * Given an array of integers. Find two disjoint contiguous sub-arrays such that the absolute difference between
+ * the sum of two sub-array is maximum.
+ * The sub-arrays should not overlap.
+ * <p/>
+ * eg- [2 -1 -2 1 -4 2 8] ans - (-1 -2 1 -4) (2 8), diff = 16
+ */
+public class SubarrayDiff {
+
+    public Answer getMaxSubarrayDiff(int[] array) {
+        int[] lmax = new int[array.length + 1];
+        int[] lmin = new int[array.length + 1];
+        int[] rmax = new int[array.length + 1];
+        int[] rmin = new int[array.length + 1];
+        lmax[array.length] = 0;
+        lmin[0] = 0;
+
+        int currentMaxSum = 0;
+        int currentMinSum = 0;
+
+        for (int i = 0; i < array.length; i++) {
+            currentMaxSum += array[i];
+            if (currentMaxSum > 0) {
+                lmax[i + 1] = Math.max(currentMaxSum, lmax[i]);
+            } else {
+                currentMaxSum = 0;
+                lmax[i + 1] = lmax[i];
+            }
+            currentMinSum += array[i];
+            if (currentMinSum < 0) {
+                lmin[i + 1] = Math.min(currentMinSum, lmin[i]);
+            } else {
+                currentMinSum = 0;
+                lmin[i + 1] = lmin[i];
+            }
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 26:41, 44:59.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.